rdr_service/tools/tool_libs/bq_res_validator.py

```diff
@@ -0,0 +1,178 @@
+            if sub == 'modules':
+                if k == 'mod_created':
+                    k2 = 'module_created'
+                elif k == 'mod_authored':
+                    k2 = 'module_authored'
+
+            if k2 not in d2:
+                not_in1.append(k2)
+                continue
+            if v != d2[k2]:
+                diff.append([k, v, d2[k2]])
+
+        if not_in1 or diff:
+            print(sub or 'Participant Summary:')
+            print('  Not In:')
+            print(f'    {not_in1}')
+            print('  Diff:')
+            print(f'    {diff}')
+
+
+    def run(self):
+        """
+        Main program process
+        :return: Exit code value
+        """
+
+        """
+        Example: Create a SQL Proxy DB connection for SQL Alchemy to use.
+
+            Calling `activate_sql_proxy()` will make a new connection to a project DB instance and
+            set the DB_CONNECTION_STRING environment var with the correct connection string for
+            SQL Alchemy.  Once the function has returned, any DAO object can be then used.
+            The connection will be closed and cleaned up when the Context Manager is released.
+        """
+        self.gcp_env.activate_sql_proxy()
+        res_gen = ParticipantSummaryGenerator()
+        bq_gen = BQParticipantSummaryGenerator()
+
+        # Stable Environment PIDs.
+        pids = [100343245, 101785380, 104831002, 104917510, 106133925, 111143764, 111693691, 111866379,
+                112094567, 113264371, 115208964, 115667157, 116772345, 118069780, 118106279, 118275699,
+                119524086, 120937445, 122162300, 123309305, 124352051, 130310274, 130332109, 133487339,
+                134691836, 135119720, 135549758, 139676702, 142593735, 143459156, 145479186, 145975075,
+                147661669, 147684969, 147942286, 147988733, 150116431, 150272201, 150540533, 150689114,
+                155241356, 157694324, 158173894, 158855869, 160498800, 163572548, 163976185, 165315114,
+                165488235, 167254162, 172505219, 175752932, 176130311, 176896382, 177254367, 181792216,
+                182766464, 183436919, 186476418, 187164690, 198860684, 201005904, 204973609, 208794758,
+                209241151, 209657895, 209658591, 211140790, 211217471, 211448524, 213605677, 214659165,
+                215839869, 216862339, 217327537, 218719201, 220866527, 225078909, 228929489, 232688732,
+                233009992, 233856454, 236294032, 236751506, 237543544, 243386095, 243643970, 243805484,
+                244855041, 245743578, 245888395, 246639275, 247220397, 248064451, 248096002, 251457975,
+                252651766, 254212435, 255107353 ]
+
+        for pid in pids:
+            print(f'PID: {pid}:')
+
+            res = res_gen.make_resource(pid)
+            res_data = res.get_data()
+
+            bq = bq_gen.make_bqrecord(pid)
+            bq_data = bq.to_dict()
+
+            self.compare_dicts(bq_data, res_data)
+
+        return 0
```

Comment:
Possible suggestion for this tool:   compare the results from the "new" bq generator to what's in the bigquery_sync  pdr_participant resource data for the  pid and confirm backwards compatibility.